File: clojure/src/project_euler/sol/p0097.clj
(ns project-euler.sol.p0097
  (:require [project-euler.lib.math :as my-math]))

(defn solve
  []
  (let [modulo (my-math/pow 10 10)]
    (format "%010d" (mod (inc (* 28433 (my-math/powermod 2 7830457 modulo))) modulo))))

File: haskell/src/Sol/P0097.hs
module Sol.P0097 (compute, solve) where

import Text.Printf (printf)

import Mylib.Math (powerModExn)

compute :: String
compute =
    printf "%010d" $ (28433 * (powerModExn 2 7830457 m) + 1) `mod` m
  where
    m = 10_000_000_000

solve :: String
solve = compute
